### DISCUSSION TITLE
opt-F3 is not working on mac for switching windows in TMUX

### DISCUSSION THREAD

**Discussion by feekApp** [ghostty-org/ghostty#3787] [Q&A]
Hi,

I noticed that `opt-F3` is not working properly for me for switching windows in TMUX sessions. In TMUX I've set the following mapping in order to switch:
```
# Quickly switch session windows
bind -n M-F1 select-window -t 1
bind -n M-F2 select-window -t 2
bind -n M-f3 select-window -t 3
bind -n M-F4 select-window -t 4
```

`opt-F1, opt-F2` and `opt-F4` are working as expected, but `opt-F3` not. In other terminals the switching using `opt-F3` is working OK! 

After running `ghostty +list-keybinds`, I do not see any binding to `F3`-key, so maybe this is a bug?

== Feek

**Comment by jmckillican** [comment]
I'm not sure if it matters, but all your Fs are uppercase except for 'f3'.  try making it uppercase and restarting ghostty?

**Comment by feekApp** [comment]
Hi @jmckillican ,

good catch, but unfortunately that was a typo in my opening post. In my actual code it is written in a capital, `F3`.

I've tried some other things, like:
```
keybind = f3=unbind
keybind = alt+f3=unbind
```
to see if there was a special macOs keybinding set in Ghostty, but no effect.

The inspector also show the right binding. strange enough the following is working:
```
keybind = alt+f3=new_window
```

after pressing `alt-F3` a new window is created! As I mentioned, in other terminals it is working as expected. So it seems like a kingd of bug.

== Feek

**Comment by feekApp** [comment]
may some worthy information, output of `sed -n l` of `opt-F1`, `opt-F2` and `opt-F3` are:
```
^[[1;3P
^[[1;3Q
^[[13;3~
```

Strange in Wezterm the code differs for the `opt-F3` key: `^[[1;3R`

== Feek

**Comment by feekApp** [comment]
Also tried `keybind = clear` for resetting all keymappings, but also no effect, so seems like the `F3`-key is is not well defined.

**Comment by feekApp** [comment]
Hi finally I managed to get back the binding as follows:
`keybind = alt+f3=esc:[1;3R`

As I mentioned before in Ghossty `opt-F3` key is assigned to: `[1;3~` so there is definitely something different / not right assigned

Should I make a bug report of this discussion?

**Comment by sespiros** [comment]
Same bug here with `Ctrl+Shift+F3` in byobu (silently ignored, while `Ctrl+Shift+F4` works).

Root cause: F3 is the only key in the F1-F4 group still using the deprecated `\x1b[13;{mod}~` encoding (see `src/input/function_keys.zig:92`).
F1, F2, and F4 were moved to the modern `\x1b[1;{mod}{P,Q,S}` form in #257 (closing #499), but F3 was missed.
That matches @feekApp's observation: `opt-F3` sends `^[[13;3~` while xterm/Wezterm send `^[[1;3R`.

The fix is two characters (`\x1b[13;{}~` → `\x1b[1;{}R`) plus the matching test update.
Only the modified CSI path changes; unmodified F3 stays on SS3 (`\x1bOR`).

I have it on a branch and verified locally (tests pass, bytes match xterm, byobu's swap-window now fires).
Could a maintainer promote this discussion to an issue so the PR has something to link?

AI disclosure: Claude Code helped me trace the inconsistency. The change is two characters and I can explain the encoding and why tmux only matches the modern form.
